apps/src/lib/levelbuilder/lesson-editor/ActivitySectionCardButtons.jsx

```diff
@@ -108,8 +109,9 @@ class ActivitySectionCardButtons extends Component {
     this.setState({addResourceOpen: true});
   };
 
-  handleCloseAddResource = () => {
+  handleCloseAddResource = resourceKey => {
     this.setState({addResourceOpen: false});
+    this.props.appendResourceLink(resourceKey);
```

Comment:
You might want to call appendResourceLink once setState has run just to be safe. Something like `this.setState({addResourceOpen: false}, this.props.appendResourceLink(resourceKey));` I think would be what you would want.